externals/kyuubi-spark-sql-engine/src/main/scala/org/apache/kyuubi/engine/spark/operation/ExecuteStatement.scala

```diff
@@ -39,13 +41,18 @@ class ExecuteStatement(
     override val statement: String,
     override val shouldRunAsync: Boolean,
     queryTimeout: Long,
-    incrementalCollect: Boolean)
+    incrementalCollect: Boolean,
+    opHandleOption: Option[OperationHandle])
   extends SparkOperation(session) with Logging {
 
+  override protected val handle: OperationHandle = opHandleOption.getOrElse(OperationHandle())
+
   private val operationLog: OperationLog = OperationLog.createOperationLog(session, getHandle)
   override def getOperationLog: Option[OperationLog] = Option(operationLog)
   override protected def supportProgress: Boolean = true
 
+  EventBus.post(SparkOperationEvent(this))
```

Comment:
it's quite tricky behavior, we'd better refactor this to avoid confusing developers, if not easy, add comments to explain what happened as least.